contracts/deployables/public-sale/PublicSaleV1.sol

```diff
@@ -474,14 +578,39 @@ contract PublicSaleV1 is
         SaleState state = saleState();
 
         if (state == SaleState.SUCCEEDED) {
-            // send the caller their purchased sale tokens
+            // sale succeeded
+
             uint256 saleTokenAmount = ($.commitments[msg.sender] * PRECISION) /
                 $.saleTokenPrice;
 
-            IERC20($.saleToken).safeTransfer(recipient_, saleTokenAmount);
+            if ($.hedgeyLockupParams.enabled) {
+                // approve hedgey lockup plan to transfer the sale token
+                IERC20($.saleToken).approve(
+                    $.hedgeyLockupParams.votingTokenLockupPlans,
+                    saleTokenAmount
+                );
+
+                // create hedgey lockup plan for the caller
+                IVotingTokenLockupPlans(
+                    $.hedgeyLockupParams.votingTokenLockupPlans
+                ).createPlan(
```

Comment:
I noticed there’s a `getLockupPlanCreatedEvent` function in the test file that retrieves the planId from the Hedgey event. Could it be that we might also need the planId on the frontend later? If so, maybe we could update the `SuccessfulSaleSettled` event to include the `planId` returned from the `createPlan` function call.

Reply:
Yes, this is actually added in PR #433.